### PR TITLE
Remove the support of the old Star-Tree

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Remove the support of the old Star-Tree generated with the StarTreeIndexSpec (already deprecated)
Remove the related metadata (rawDoc and aggregatedDoc)
Remove the field size and derived metric type from MetricFieldSpec
Simplify the segment generation logic

Backward-incompatible: (checked with the community to make sure there is no use case with the old Star-Tree)
Existing segments with old Star-Tree won't work properly

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/stats/StringColumnPreIndexStatsCollector.java [hunk 1]
```diff
@@ -26,124 +26,82 @@
...
+        _maxLength = Math.max(_maxLength, length);
       }
-      updateTotalNumberOfEntries((Object[]) entry);
-    } else {
 
-      String value;
-      if (entry != null) {
-        value = entry.toString();
-      } else {
-        value = fieldSpec.getDefaultNullValue().toString();
```
Comment (siddharthteotia) [inline pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/stats/StringColumnPreIndexStatsCollector.java]: Why this change of not using getDefaultNullValue()?

Comment (siddharthteotia) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/stats/StringColumnPreIndexStatsCollector.java]: Ignore. Got it.... this is STRING column

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/stats/StringColumnPreIndexStatsCollector.java]: Null values are handled by NullValueTransformer. Here we will never get null entries.
